### PR TITLE
Update docker versions; Fix #4198

### LINKED ISSUE #4198: Improve docker image performance
- Switch databases to in-memory
- Remove fsync
- Activate async I/O

### PR DESCRIPTION
**Pull request by mlassnig**

### CODE REVIEW

**Comment by maany** [comment]
LGTM! It is important to merge this quickly as it contains changes required in `rucio/rucio` for fixing `test-conveyor` and `test-tpc` since the `fts-rest-client` was updated in https://github.com/rucio/containers/pull/182

**Review by maany — COMMENTED** [review]
LGTM, just the header in  `lib/rucio/tests/test_tpc.py` needs to be fixed. ;)

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -41,39 +41,6 @@ jobs:
...
-
-          cd $GITHUB_WORKSPACE
-
-          # Check if rucio/containers has a corresponding $GIT_REF 
-          if [ $(git tag -l "$GIT_REF") ]; then
-              git checkout tags/$GIT_REF
-          else
-              echo "Tag $GIT_REF not found in rucio/containers. Integration test containers will be built against the master branch instead."
-              git checkout master
-          fi
```
Comment (joeldierkes) [inline .github/workflows/integration_tests.yml]: Why is this deleted?
As far as I know, this was included (#5256) since changes made to the rucio/containers repo were only used when a new container was released. However, we want to use these changes in the tests immediately.

Comment (maany) [reply .github/workflows/integration_tests.yml]: It needs to be patched. I will send a separate PR later today.  ;)

File: etc/docker/test/matrix.yml [hunk 2]
```diff
@@ -40,9 +40,9 @@ suites:
       - oracle
       - mysql8
       - postgres9
-      - postgres12
+      - postgres14
```
Comment (bari12) [inline etc/docker/test/matrix.yml:43]: I think this is good, I am wondering though about the `postgres9` one. In principle this is not supported anymore since end of last year. I wonder if we should just test against postgres14 and remove postgres9 entirely.
Raises a bit also the question of our compatibilities we mention in http://rucio.cern.ch/documentation/requirements

Comment (mlassnig) [reply etc/docker/test/matrix.yml:43]: We only have the old one because psql9 is the only available one in CentOS7. psql10 is default in CentOS8, however psql12 is also available. I think it's fine to just have psql14.

Comment (bari12) [reply etc/docker/test/matrix.yml:43]: As discussed, let's remove everything but postgres14

**Comment by bari12** [comment]
Rebase please

**Review by bari12 — APPROVED** [review]
